jobserver/models.py

```diff
@@ -107,45 +141,58 @@ def __str__(self):
     def get_absolute_url(self):
         return reverse("job-detail", kwargs={"pk": self.pk})
 
-    @property
-    def is_complete(self):
-        return self.status_code is None and self.completed_at
-
-    @property
-    def is_dependency_failed(self):
-        return self.status_code == 7
-
     @property
     def is_failed(self):
-        dependency_not_finished = 6
-        dependency_failed = 7
-        dependency_running = 8
-
-        non_failure_status = self.status_code in [
-            dependency_not_finished,
-            dependency_failed,
-            dependency_running,
+        if not self.is_finished:
+            return False
+
+        non_failure_statuses = [
+            STATE_DEPENDENCY_NOT_FINISHED,
+            STATE_DEPENDENCY_RUNNING,
+            STATE_SUCCESS,
         ]
 
-        return self.status_code and not non_failure_status
+        return self.status_code not in non_failure_statuses
 
     @property
-    def is_in_progress(self):
-        if self.status_code is not None:
-            return False
-
-        return self.started_at and not self.completed_at
+    def is_finished(self):
+        # no status code set
+        return (
+            self.started
+            and self.status_code is not None
+            and self.completed_at is not None
+        )
 
     @property
     def is_pending(self):
-        if self.status_code in [6, 8]:
+        if not self.started:
             return True
 
-        if self.status_code is None and not (self.started_at and self.completed_at):
+        pending_states = [
+            STATE_DEPENDENCY_NOT_FINISHED,
+            STATE_DEPENDENCY_RUNNING,
+        ]
+
+        if self.status_code in pending_states:
             return True
 
         return False
 
+    @property
+    def is_running(self):
+        # can we rely on status_code being null to define running?
+        if self.status_code and self.status_code not in [6, 8]:
```

Comment:
Can we use the module-scoped variables here?

Reply:
Oops, missed those, yes!